src/components/MainPage.js

```diff
@@ -0,0 +1,140 @@
+import * as MoviesActions from '../actions/actions';
+import PropTypes from 'prop-types';
+
+const styles = () => ({
+  root: {
+    flexGrow: 1,
+  },
+  textField: {
+    width: 445,
+  },
+  card: {
+    maxWidth: 445,
+    margin: '20px auto',
+  },
+  media: {
+    height: 0,
+    paddingTop: '100%',
+  },
+});
+
+class MainPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      isLoaded: false,
+      inputValue: '',
+    };
+  }
+
+  componentDidMount() {
+    this.props.actions.fetchMovies();
+    this.setState({
+      isLoaded: true,
+    });
+  }
+
+  handleSearchChange = event => {
+    this.setState({ inputValue: event.target.value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    this.props.actions.searchMovies(this.state.inputValue);
+  };
+
+  render() {
+    const { error, isLoaded } = this.state;
+    const { classes } = this.props;
+    const data = this.props.data;
+    const poster = 'https://image.tmdb.org/t/p/w500';
+
+    if (error) {
+      return <div>Error: {error.message}</div>;
+    } else if (!isLoaded) {
+      return <div>Is loading...</div>;
+    } else {
+      return (
+        <Fragment>
+          <Fragment>
+            <form onSubmit={this.handleSubmit}>
+              <TextField
+                id="search"
+                onChange={this.handleSearchChange}
+                label="Search field"
+                type="search"
+                className={classes.textField}
+                margin="normal"
+              />
+            </form>
+          </Fragment>
+          {data.map(item => (
+            <div key={item.id}>
+              <Card className={classes.card}>
+                <CardMedia
+                  className={classes.media}
+                  image={`${poster}${item.poster_path}`}
+                  alt={item.title}
+                  title={item.title}
+                />
+                <CardContent>
+                  <Typography gutterBottom variant="headline" component="h2">
+                    {item.title}
+                  </Typography>
+                  <Typography component="p">{item.overview}</Typography>
+                </CardContent>
+                <CardActions>
+                  <Button size="small" color="primary">
+                    Share
+                  </Button>
+                  <Button size="small" color="primary">
+                    Learn More
+                  </Button>
+                </CardActions>
+              </Card>
+            </div>
+          ))}
+        </Fragment>
+      );
+    }
+  }
+}
+
+MainPage.propTypes = {
+  actions: PropTypes.object,
+  data: PropTypes.array,
+};
+
+function mapStateToProps(state) {
+  if (state.searchMovies.length > 0) {
+    return { data: state.searchMovies };
+  } else if (state.defaultMovies) {
+    return { data: state.defaultMovies };
+  }
+}
+
+function mapDispatchToProps(dispatch) {

```

Comment:
I think better will be if you use just use an object here:

```js
const mapDispathToProps = {
  /* list of actions */
}
```

In such way you will see what concrete actions passed to the container